library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -823,8 +822,15 @@ public void updateChannel(@NonNull Channel channel, @Nullable Message updateMess
         onSetUserCompleted(new ClientConnectionCallback() {
             @Override
             public void onSuccess(User user) {
-                apiService.updateChannel(channel.getType(), channel.getId(), apiKey, clientID,
-                        new UpdateChannelRequest(channel.getExtraData(), updateMessage))
+                Map<String, Object> body = new HashMap<>();
+                // check reserved field
+                channel.getExtraData().remove("members");
+                body.put("data",channel.getExtraData());
+                if (updateMessage != null) {
+                    String str = GsonConverter.Gson().toJson(updateMessage);
+                    body.put("message", GsonConverter.Gson().fromJson(str, Map.class));
+                }
+                apiService.updateChannel(channel.getType(), channel.getId(), apiKey, clientID, body)
```

Comment:
Replacing class with a HashMap is code degradation within strongly type language.

If Retrofit should keep nulls in json we need to call `serializeNulls` when setting Gson converter. If it's not global API requirement, then we need to ask backend developers why `Dangling name: message` is returned. API should be consistent, otherwise we will catch such king of issues always.